### ISSUE TITLE
Permission denied while trying to cleanup previous running result

### ISSUE THREAD

**Issue by lanking520** [actions/setup-java#365]
**Description:**
Setup Java V3 failed during the setup period on self-hosted runner:

```
Run actions/setup-java@v3
  with:
    distribution: corretto
    java-version: 11
    java-package: jdk
    architecture: x64
    check-latest: false
    server-id: github
    server-username: GITHUB_ACTOR
    server-password: GITHUB_TOKEN
    overwrite-settings: true
    job-status: success
Trying to resolve the latest version from remote
Resolved latest version as 11.0.16.8.1
Trying to download...
Downloading Java 11.0.16.8.1 (Corretto) from https://corretto.aws/downloads/resources/11.0.16.8.1/amazon-corretto-11.0.16.8.1-linux-x64.tar.gz ...
Extracting Java archive...
/bin/tar xz --warning=no-unknown-keyword -C /home/ubuntu/actions-runner/_work/_temp/04d844[2](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:2)c-5c58-4ef2-b0ce-daa[3](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:3)1[4](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:4)0432c3 -f /home/ubuntu/actions-runner/_work/_temp/2e97ea69-84be-49[5](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:5)0-8014-8d[6](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:6)[7](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:7)2f0f1d70
Error: Command failed: rm -rf "/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.[8](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:8).1/x64"
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/[11](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:11).0.[16](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:17).8.1/x64/bin/pack[20](https://github.com/deepjavalibrary/djl-serving/runs/7771831417?check_suite_focus=true#step:3:21)0': Permission denied
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.8.1/x64/bin/serialver': Permission denied
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.8.1/x64/bin/jinfo': Permission denied
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.8.1/x64/bin/java': Permission denied
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.8.1/x64/bin/jlink': Permission denied
rm: cannot remove '/home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/11.0.16.8.1/x64/bin/jmod': Permission denied
```

**Comment by panticmilos** [comment]
hi @lanking520, thank you for submitting the issue. To start which self-hosted runner are you using? Also, could you link us repo where the issue is or provide the minimum needed repro steps repo?

**Comment by dmitry-shibanov** [comment]
Hello @lanking520 just a gentle ping.

**Comment by lanking520** [comment]
I think this might leads to our Machine issue. This is the yml we use: https://github.com/deepjavalibrary/djl-serving/blob/master/.github/workflows/integration.yml#L190.

**Comment by lanking520** [comment]
I am cleaning up the stuff under _work manually

**Comment by panticmilos** [comment]
Hi @lanking520,

I have tried [running](https://github.com/panticmilos/djl-serving/actions/runs/3083477531/jobs/4984452516) your jobs on the hosted machine, and this problem doesn't seem to appear. This indicates the issue might be in the self-hosted runner.  I was going to suggest you try to use `sudo` command, but I saw you have already tried it. It looks like a permission issue to me.  (By the way, some of the jobs did fail in my reconstruction, but a step that is the issue here didn't)

Since this issue seems to be related to the self-hosted runner, for now, I am going to close it. Feel free to continue the conversation here, reopen it, or create a new issue. Cheers
